### PR TITLE
Feat: Conditional submit based on gas price

### PR DESCRIPTION
**Pull request by annieke**
Addresses https://github.com/ethereum-optimism/roadmap/issues/540

### CODE REVIEW

File: src/batch-submitter/batch-submitter.ts [hunk 1]
```diff
@@ -11,7 +10,7 @@ import { getContractFactory } from '@eth-optimism/contracts'
 
 /* Internal Imports */
 import { Address, Bytes32 } from '../coders'
-
+import { gasInGwei } from '../utils'
```
Comment (tynes) [inline src/batch-submitter/batch-submitter.ts:13]: Creating this helper function is a good idea, there is also an ethers helper function that does this too - https://docs.ethers.io/v5/api/utils/display-logic/

File: src/batch-submitter/tx-batch-submitter.ts [hunk 2]
```diff
@@ -211,6 +214,12 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     startBlock: number,
     endBlock: number
   ): Promise<TransactionReceipt> {
+    // Do not submit batch if gas price above threshold
+    const gasPriceInGwei = gasInGwei(await this.signer.getGasPrice())
+    if (gasPriceInGwei > this.gasThresholdInGwei) {
+      return
```
Comment (tynes) [inline src/batch-submitter/tx-batch-submitter.ts:220]: Could we do a `log.info` here with something along the lines of:
`Current gas price ${gasPriceInGwei} is higher than gas price threshold ${gasThresholdInGwei}`

**Comment by tynes** [comment]
The tests look good to me! Ready for merge after adding the logline

File: src/batch-submitter/batch-submitter.ts (hunk 1 above)
Comment (annieke) [reply src/batch-submitter/batch-submitter.ts:13]: migrating to this to minimize code we maintain!

File: src/batch-submitter/tx-batch-submitter.ts [hunk 3]
```diff
@@ -211,6 +213,16 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
     startBlock: number,
     endBlock: number
   ): Promise<TransactionReceipt> {
+    // Do not submit batch if gas price above threshold
+    const gasPriceInGwei = parseInt(
+      ethers.utils.formatUnits(await this.signer.getGasPrice(), 'gwei'), 10
+    )
+    if (gasPriceInGwei > this.gasThresholdInGwei) {
+      this.log.info(`Current gas price ${gasPriceInGwei} is higher 
+                      than gas price threshold ${this.gasThresholdInGwei}`)
```
Comment (tynes) [inline src/batch-submitter/tx-batch-submitter.ts:222]: This is going to result in a `/n` in the logline itself. There are two ways to fix it - concatenate the two strings with `+` or put the newline directly after `this.log.info(`

**Comment by tynes** [comment]
@annieke The integration tests are passing again!

File: src/batch-submitter/tx-batch-submitter.ts (hunk 3 above)
Comment (annieke) [reply src/batch-submitter/tx-batch-submitter.ts:222]: ooo completely forgot abt this, fixing

**Review by tynes — APPROVED** [review]
Looks good to me!

**Pull request merged**